lm_eval/__main__.py

```diff
@@ -19,7 +19,9 @@ def _handle_non_serializable(o):
         return int(o)
     elif isinstance(o, set):
         return list(o)
-    raise TypeError(f"Object of type {o.__class__.__name__} is not JSON serializable")
+    else:
```

Comment:
In qasper_freeform, I found the output of the result contained some class.function which may cause errors when saving to json
<img width="1092" alt="image" src="https://github.com/EleutherAI/lm-evaluation-harness/assets/28424280/aef5ac0d-8b1e-4f58-8efe-12574278b1cd">

Reply:
The implementation before your PR handles that part. You could include another option the handler where the conversion becomes str.